packages/client/src/blockchain/chain.ts

```diff
@@ -297,13 +297,13 @@ export class Chain {
     headers.latest = await this.getCanonicalHeadHeader()
     // finalized and safe are always blocks since they have to have valid execution
     // before they can be saved in chain
-    headers.finalized = (await this.getCanonicalFinalizedBlock()).header
-    headers.safe = (await this.getCanonicalSafeBlock()).header
+    headers.finalized = blocks.finalized?.header ?? null
+    headers.safe = blocks.safe?.header ?? null
     headers.vm = (await this.getCanonicalVmHead()).header
 
     blocks.latest = await this.getCanonicalHeadBlock()
```

Comment:
Ok, so in this file if one first does gets all the blocks and then the headers, then a sync test fails. If one updates this to the original (see 586868ba51d9d3beddd5961fa44d667c4d547d8d ) then it passes. Test in question is `integration/lightclient.spec.ts`